### PR TITLE
Fix some collisions above/below staff (fixes #1634, half of #1633)

### LINKED ISSUE #1634: Missing abovelinestextheight in nabc
If a score has nabc, the abovelinestextheight is sometimes missing. Adding, for example, a low note causes abovelinestextheight to appear.

```
\documentclass{article}
\usepackage{gregoriotex}
\begin{document}

\long\def\nabcsnippet#1{%
  \directlua{gregoriotex.direct_gabc("\luatexluaescapestring{\unexpanded\expandafter{#1}}", "nabc-lines:1;", false)}%
}%

\gresetinitiallines{0}

{\grechangedim{abovelinestextheight}{1in}{scalable}%

abovelinestextheight=1in missing:
\nabcsnippet{(c3) g|pu(g|pu)}%

abovelinestextheight=1in present:
\nabcsnippet{(c3) a|pu(a|pu)}%
}
\end{document}
```

Image

### PR DESCRIPTION
**Pull request by davidweichiang**
The C code was telling the Lua code when to compute line heights, and it was missing some cases, causing not enough additional space to be added. Now the Lua code always computes line heights (which is simpler).

This breaks several tests: clef_change.gabc and nabc-edge-cases.gabc because the old results had bugs, but fix-569 because it uncovers a new bug (https://github.com/gregorio-project/gregorio/issues/1636).

The fact that the C code does not always know when additional space is needed points to the fact that uniform line heights probably have several bugs (i.e., whenever \GreGlyphHeights is called from inside the TeX code). The first half of https://github.com/gregorio-project/gregorio/issues/1633 is one of these cases, but I haven't tested the others.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ As of v3.0.0 this project adheres to [Semantic Versioning](http://semver.org/).
 - Fixed the placement of above-lines text (`<alt>`) relative to a note above the staff or when the number of staff lines is not 4. See [#1613](https://github.com/gregorio-project/gregorio/issues/1613) and [#1614](https://github.com/gregorio-project/gregorio/issues/1614).
 - Fixed a bug where the above-lines text (`<alt>`) could collide with a note above the staff. See [#1613](https://github.com/gregorio-project/gregorio/issues/1613).
 - Fixed a bug that could cause the clef and staff to be printed too high. See [#1503](https://github.com/gregorio-project/gregorio/issues/1503).
+- Fixed a bug that could cause insufficient additional space to be added above or below the staff. See [#1633](https://github.com/gregorio-project/gregorio/issues/1633) and [#1634](https://github.com/gregorio-project/gregorio/issues/1634).
```
Comment (eschwab) [inline CHANGELOG.md]: This entry should be in the Unreleased section above. Under a subheading `### Fixed`

**Comment by davidweichiang** [comment]
Sorry I really messed up the rebase. It should be fixed now.
